src/Umbraco.Web.UI.Client/src/common/services/tree.service.js

```diff
@@ -287,7 +287,10 @@ function treeService($q, treeResource, iconHelper, notificationsService, eventsS
                 throw "No node defined on args object for loadNodeChildren";
             }
 
-            this.removeChildNodes(args.node);
+            // don't remove the children for container nodes in dialogs, as it'll remove the right arrow indicator
```

Comment:
Isn't this just a symptom of a different problem?

Reply:
Well... it's a workaround for because `getChildren()` never returns any children for container nodes due to performance considerations. You can say it's a symptom or a derivative effect of mixing two navigation paradigms (tree and list view) in one navigation.

I would love to let `getChildren()` return something else for container nodes, but I'm afraid it'll break too many things at this point.